creed/rosetta.py

```diff
@@ -0,0 +1,290 @@
+#     fov_num = raw_images.shape[0]
+#     # compute difference between first compensation image and baseline
+#     diff_images = raw_images.values - comp_dict[comp_dir_list[0]].values
+#     diff_images[diff_images < 0] = 0
+#
+#     comp_image_num = len(comp_dir_list)
+#     # loop over each channel
+#     for j in range(raw_images.shape[3]):
+#         # create tiled array of corrections x fovs
+#         tiled_image = np.zeros((img_size * (comp_image_num + 2),
+#                                 img_size * fov_num))
+#         # loop over each fov, and place into columns of tiled array
+#         for i in range(raw_images.shape[0]):
+#             start = i * img_size
+#             end = (i + 1) * img_size
+#
+#             # first row is raw image
+#             tiled_image[:img_size, start:end] = raw_images.values[i, :, :, j]
+#
+#             # second row is difference image between raw and first compensated image
+#             tiled_image[img_size:(img_size * 2), start:end] = diff_images[i, :, :, j]
+#
+#             # subsequent rows are compensated images with different coefficients
+#             for idx, key in enumerate(comp_dir_list):
+#                 tiled_image[(img_size * (idx + 2)):(img_size * (idx + 3)), start:end] = \
+#                     comp_dict[key].values[i, :, :, j]
+#
+#         io.imsave(os.path.join(output_dir, raw_images.channels.values[j] + '_comparison.tiff'),
+#                   tiled_image)
+
+
+def create_tiled_comparison(input_dir_list, output_dir):
+    """Creates a tiled image comparing FOVs from all supplied runs for each channel.
+
+    Args:
+        input_dir_list: list of directories to compare
+        output_dir: directory where tifs will be saved"""
+
+    # load images
+    dir_dict = {}
+    dir_shapes = []
+    for dir_name in input_dir_list:
+        dir_images = load_imgs_from_tree(dir_name)
+        dir_shapes.append(dir_images.shape)
+        dir_dict[dir_name] = dir_images
+
+    if not np.all([shape == dir_shapes[0] for shape in dir_shapes]):
+        raise ValueError("All directories must contain the same number of images")
+
+    first_dir = dir_dict[input_dir_list[0]]
+    img_size = first_dir.shape[1]
+    fov_num = first_dir.shape[0]
+
+    # loop over each channel
+    for j in range(first_dir.shape[3]):
+        # create tiled array of dirs x fovs
+        tiled_image = np.zeros((img_size * len(input_dir_list),
+                                img_size * fov_num), dtype=first_dir.dtype)
+
+        # loop over each fov, and place into columns of tiled array
+        for i in range(first_dir.shape[0]):
+            start = i * img_size
+            end = (i + 1) * img_size
+
+            # go through each of the directories and place in appropriate spot
+            for idx, key in enumerate(input_dir_list):
+                tiled_image[(img_size * idx):(img_size * (idx + 1)), start:end] = \
+                    dir_dict[key].values[i, :, :, j]
```

Comment:
Think we can remove `dir_dict` completely and simply do:

```
for idx, key in enumerate(input_dir_list):
    dir_images = load_imgs_from_tree(key)
    tiled_image[(img_size * idx):(img_size * (idx + 1)), start:end] = dir_images
```

Reply:
This would require us to reload the entire directory for every iteration of the for loop. I agree it looks simpler, but the performance hit here would be significant, especially for larger directories

Reply:
Ah right, got a bit ahead of myself with this one.